doozer/doozerlib/cli/scan_sources.py

```diff
@@ -54,6 +68,143 @@ def run(self):
         # We have our information. Now build and print the output report
         self.generate_report()
 
+    def _try_reconciliation(self, metadata: Metadata, repo_name: str, branch_name: str):
+        reconciled = False
+
+        # Attempt a fast-forward merge
+        rc, _, _ = exectools.cmd_gather(cmd=['git', 'pull', '--ff-only', 'public_upstream', branch_name])
+        if not rc:
+            # fast-forward succeeded, will push to openshift-priv
+            self.runtime.logger.info('Fast-forwarded %s from public_upstream/%s', metadata.name, branch_name)
+            reconciled = True
+
+        else:
+            # fast-forward failed, trying a merge commit
+            rc, _, _ = exectools.cmd_gather(cmd=['git', 'merge', f'public_upstream/{branch_name}',
+                                                 '-m', f'Reconciliation with public openshift/{repo_name}'],
```

Comment:
Well, the public org name may not be openshift. Let's be more accurate.

Reply:
Done with aa34afcf3f39807ee8296454145cfc50ca846907

I chose to apply a minimal change, and let the user find the public upstream if interested. In most cases (if not all), this will be `openshift` so I think it's an acceptable compromise to minimize changes.